compiler/rustc_resolve/src/late/lifetimes.rs

```diff
@@ -1407,6 +1405,32 @@ impl<'a, 'tcx> Visitor<'tcx> for LifetimeContext<'a, 'tcx> {
                     }) => {
                         this.visit_lifetime(lifetime);
                         walk_list!(this, visit_param_bound, bounds);
+
+                        if lifetime.name != hir::LifetimeName::Static {
+                            for bound in bounds {
+                                let hir::GenericBound::Outlives(ref lt) = bound else {
+                                    continue;
+                                };
+                                if lt.name != hir::LifetimeName::Static {
+                                    continue;
+                                }
+                                this.insert_lifetime(lt, Region::Static);
+                                this.tcx
+                                    .sess
+                                    .struct_span_warn(
+                                        lifetime.span,
```

Comment:
Add a FIXME so we'll try to use the span of the definition of the lifetime instead of the use

Reply:
Commit 0b8a80b3e9db1181eebaf246df0f634a64179083 does that. I'm not convinced by the diagnostic: the span is a bit far from the reason it's generated.

Reply:
OK yea, that's worse. Maybe we'll need to create a multi span diagnostic or just a suggestion that removes the param and renames all uses to 'static.

Anyway, off-topic